tendrl/node_agent/manager/manager.py

```diff
@@ -240,6 +242,27 @@ def on_pull(self, raw_data):
                     self.etcd_client.write(("nodes/%s/Disks/free/%s") % (
                         raw_data["node_id"], disk), "")
 
+    def load_and_execute_sds_discovery_plugins(self):
+        # Execute the SDS discovery plugins and tag the nodes with data
+        for plugin in SDSDiscoveryManager().get_available_plugins():
+            sds_details = plugin.discover_storage_system()
+            if len(sds_details.keys()) > 0:
+                dict = {}
+                for key in sds_details['cluster_attrs'].keys():
+                    dict[key] = sds_details['cluster_attrs'][key]
+                self.persister_thread.update_node_context(
```

Comment:
Please note that line 151 of this file will override all these details if there is any change to hardware inventory. Please fix this